superset/assets/src/datasource/DatasourceEditor.jsx

```diff
@@ -565,18 +569,22 @@ export class DatasourceEditor extends React.PureComponent {
   }
 
   render() {
-    const datasource = this.state.datasource;
+    const { datasource, activeTabKey } = this.state;
     return (
       <div className="Datasource">
         {this.renderErrors()}
         <Tabs
           id="table-tabs"
           onSelect={this.handleTabSelect}
-          defaultActiveKey={1}
+          defaultActiveKey={activeTabKey}
         >
           <Tab eventKey={1} title={t('Settings')}>
-            {this.state.activeTabKey === 1 &&
+            {activeTabKey === 1 &&
               <div>
+                <div className="change-warning well">
+                  <span>Be careful</span>. Changes to these settings can break all charts,
```

Comment:
This should all be a `t()` string. That means we'd lose the bold for part of it, but I think it's ok. If you really want `Be careful` to be in bold, then you could format it that way as a title string or something like that. Also, I'd probably reword/reformat as this (since changes won't always break charts): (pseudocode)
```jsx
<div>
  <div className="bold">{t('Warning')}</div>
  <div>{t('Changing these settings will affect all charts using this datasource, including charts owned by other people.')}</div>
</div>
```

Reply:
fixed. I prefer to keep all warning message in single line, so that save some vertical space for Settings content in modal.